src/api/chrome.js

```diff
@@ -215,3 +218,32 @@ const getBadgeCounter = async (initialValue = 0) => {
   }
   return badgeCounter
 }
+
+/**
+ * Retrieves current tab url and validates it. Returns valid tab url otherwise undefined.
+ * @returns the tab url or undefined
+ */
+
+export const getCurrentTabUrlBackground = async () => {
+  let queryOptions = { active: true }
+  const currWindow = await chrome.windows.getCurrent()
+  // `tab` will either be a `tabs.Tab` instance or `undefined`.
+  let [tab] = await chrome.tabs.query({
+    ...queryOptions,
+    windowId: currWindow.id
+  })
+
+  if (!isValidUrl(tab.url, URL_MATCH)) {
+    return undefined
+  }
+  return tab.url
+}
+
+/**
+ * Sends a message with proper action to the background service and returns its response.
+ * @returns the current tab url or undefined
+ */
+export const getCurrentTabUrl = async () => {
+  const tabUrl = await sendMessage(ACTION_GET_CURRENT_TAB_URL)
+  return tabUrl
+}
```

Comment:
unnecessary code

```suggestion
export const getCurrentTabUrl = () => {
  return sendMessage(ACTION_GET_CURRENT_TAB_URL)
}
```

Reply:
I wanted to be more explicit but I can remove redundant variable.  I'd stay though with the `async` at least. wdyt?